mutiny-core/src/lib.rs

```diff
@@ -2286,6 +2290,60 @@ impl<S: MutinyStorage> MutinyWallet<S> {
     pub fn is_safe_mode(&self) -> bool {
         self.safe_mode
     }
+
+    /// Calls upon a Cashu mint and withdrawls/redeem/melts the token from it.
+    pub async fn melt_cashu_token(
+        &self,
+        token_v3: TokenV3,
+    ) -> Result<Vec<MutinyInvoice>, MutinyError> {
+        let mint_client = CrossPlatformHttpClient::new();
+        let mut invoices: Vec<MutinyInvoice> = Vec::with_capacity(token_v3.tokens.len());
+
+        for token in token_v3.tokens {
+            let mint_url = match token.mint {
+                Some(url) => url,
+                None => return Err(MutinyError::EmptyMintURLError),
+            };
+
+            let invoice_amount = token.proofs.total_amount() as f64 * (1.0 - 3.0 / 100.0);
+
+            let mut mutiny_invoice = self
+                .create_invoice(invoice_amount as u64, vec![MELT_CASHU_TOKEN.to_string()])
+                .await?;
+
+            let mutiny_invoice_str = mutiny_invoice
+                .bolt11
+                .clone()
+                .expect("The invoice should have BOLT11");
+
+            let melt_quote_res = mint_client
+                .post_melt_quote_bolt11(
+                    &mint_url,
+                    mutiny_invoice_str.to_string(),
+                    CurrencyUnit::Sat,
+                )
+                .await?;
+
+            let post_melt_bolt11_response = mint_client
+                .post_melt_bolt11(&mint_url, token.proofs, melt_quote_res.quote, vec![])
+                .await?;
+            let amt: u64 = post_melt_bolt11_response
+                .change
+                .iter()
+                .map(|s| s.amount)
+                .sum();
+            log_debug!(
+                self.logger,
+                "will need to add extra outputs for {amt} for overpaid fees"
+            );
```

Comment:
I'm a little confused why this debug is here. Should this be a warning? And does it always occur?

Reply:
@benthecarman commented to add that log that could be for a follow-up. Follow up could be to add outputs param to the `post_melt_bolt` call in order to get the overpaid fees if any. Or if we create an invoice for "right" amount, we could get rid of this.